jekyll/_cci2/configuration-reference.adoc

```diff
@@ -2728,19 +2733,128 @@ workflows:
       - build_server_pdfs: # << the job to conditionally run based on the filter-by-branch-name.
           filters:
             branches:
-              only: /server\/.*/
+              only: /server\/.*/ # the job build_server_pdfs will only run when the branch being built starts with server/
 ----
 
-The above snippet causes the job  `build_server_pdfs` to only be run when the branch being built starts with "server/".
+You can read more about using regular expressions in your config in the xref:workflows#using-regular-expressions-to-filter-tags-and-branches[Using workflows to schedule jobs] page.
+
+'''
+
+====== Expression-based job filters
+Expression-based job filters allow you to conditionally run jobs based on the following:
+
+* xref:variables#pipeline-values[Pipeline values]
+* xref:pipeline-variables#pipeline-parameters-in-configuration[Pipeline parameters]
 
-You can read more about using regex in your config in the xref:workflows#using-regular-expressions-to-filter-tags-and-branches[Using workflows to schedule jobs] page.
+An expression-based job filter is a rule that is evaluated against pipeline values and parameters to decide whether a job should run.
+
+Using expression-based job filters is one way to optimize your pipelines to lower costs, decrease time to feedback, or run specific jobs based on the context of the source of change.
+
+[,yml]
+----
+workflows:
+  deploy:
+    jobs:
+      - init-service
+      - build-service-image:
+          requires:
+            - init-service
+      - dry-run-service:
+          requires:
+            - init-service
+          filters: pipeline.git.branch != "main" and pipeline.git.branch != "canary"
+      - publish-service:
+          requires:
+            - build-service-image
+            - test-service
+          filters: pipeline.git.branch == "main" or pipeline.git.tag starts-with "release"
+      - deploy-service:
+          context:
+            - org-global
+          requires:
+            - publish-service
+          filters: pipeline.git.branch == "main" and pipeline.git.commit.subject starts-with "DEPLOY:"
+----
+
+'''
+
+**Examples**
+
+Allow the job to run only on the project's `main` branch:
```

Comment:
~ What do you think about

> Only run the job on the project's `main` branch:

I feel we should avoid using words like "allow" or "deny" because they imply a permission control but since anybody can modify the filters for commits the push on an unprotected branch they can always make the job run if they want.

Reply:
Oh yeah agreed. This language is there due to being copied over from context restrictions 🙈